### PR TITLE
remorph metrics logging for usage for experimental functions

### PR DESCRIPTION
**Pull request by ganeshdogiparthi-db**

### CODE REVIEW

File: src/databricks/labs/remorph/transpiler/execute.py [hunk 1]
```diff
@@ -229,6 +229,8 @@ def verify_workspace_client(workspace_client: WorkspaceClient) -> WorkspaceClien
         workspace_client.config._product = "remorph"
     if workspace_client.config._product_version != __version__:
         workspace_client.config._product_version = __version__
+
+    logger.info(f"user: {workspace_client.current_user.me().user_name}")
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/transpiler/execute.py:233]: can you remove that from CLI so we don't log twice.

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/transpiler/execute.py (hunk 1 above)
Comment (ganeshdogiparthi-db) [reply src/databricks/labs/remorph/transpiler/execute.py:233]: I have checked @sundarshankar89 
Transpile:
In cli.py, we don't call for `verify_workspace_client`, so this is logged only once.
We call `verify_workspace_client` in experimental functions, which are not related to CLI, so this shouldn't be a problem.

Reconcile:
In reconcile we log the `user` in CLI and trigger the job. 
In the workflow job, while running the execute.py/recon function, we log it in the workflow job logs, so it does not log twice.

Please let me know if you still want me to remove it from reconcile CLI.

Comment (ganeshdogiparthi-db) [reply src/databricks/labs/remorph/transpiler/execute.py:234]: To follow up on your suggestion, I have removed the logger from cli.py/transpile and called `verify_workspace_client` to execute.py/morph function. This should cover the CLI and any cx wrapper around the morph function.

File: src/databricks/labs/remorph/transpiler/execute.py [hunk 2]
```diff
@@ -155,6 +155,7 @@ def morph(workspace_client: WorkspaceClient, config: MorphConfig):
     :param config: The configuration for the morph operation.
     :param workspace_client: The WorkspaceClient object.
     """
+    ws_client: WorkspaceClient = verify_workspace_client(workspace_client)
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/transpiler/execute.py:158]: nit: Add a comment why we are reassigning the wsclient.

**Review by sundarshankar89 — APPROVED** [review]
LGTM small nit

File: src/databricks/labs/remorph/transpiler/execute.py (hunk 2 above)
Comment (ganeshdogiparthi-db) [reply src/databricks/labs/remorph/transpiler/execute.py:158]: Done

File: .github/scripts/setup_spark_remote.sh [hunk 3]
```diff
@@ -34,4 +34,4 @@ if [ $? -ne 0 ]; then
     #sleep time to start the server
     sleep 2m
     echo "Server Already Running"
-fi
+fi
```
Comment (sundarshankar89) [inline .github/scripts/setup_spark_remote.sh:37]: remove this space so there is no change here.

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: .github/scripts/setup_spark_remote.sh (hunk 3 above)
Comment (ganeshdogiparthi-db) [reply .github/scripts/setup_spark_remote.sh]: done

File: pyproject.toml [hunk 4]
```diff
@@ -49,7 +49,7 @@ dependencies = [
   "pytest-cov>=4.0.0,<5.0.0",
   "black>=23.1.0",
   "ruff>=0.0.243",
-  "databricks-connect",
+  "databricks-connect==15.1",
```
Comment (nfx) [inline pyproject.toml:52]: don't attach specific databricks connect version. there

**Pull request merged**